packages/transactions/src/new-signatures.ts

```diff
@@ -0,0 +1,96 @@
+import { Address, getAddressFromPublicKey } from '@solana/addresses';
+import { Decoder } from '@solana/codecs-core';
+import { getBase58Decoder } from '@solana/codecs-strings';
+import {
+    SOLANA_ERROR__TRANSACTION__FEE_PAYER_SIGNATURE_MISSING,
+    SOLANA_ERROR__TRANSACTION__SIGNATURES_MISSING,
+    SolanaError,
+} from '@solana/errors';
+import { Signature, SignatureBytes, signBytes } from '@solana/keys';
+
+import { CompilableTransaction } from './compilable-transaction';
+import { compileMessage } from './message';
+import { OrderedMap } from './ordered-map';
+import { getCompiledMessageEncoder } from './serializers';
+
+export type TransactionMessageBytes = Uint8Array & { readonly __brand: unique symbol };
+
+export interface INewTransactionWithSignatures {
+    messageBytes: TransactionMessageBytes;
+    signatures: OrderedMap<Address, SignatureBytes | null>;
+}
+
+export interface INewFullySignedTransaction extends INewTransactionWithSignatures {
+    readonly __brand: unique symbol;
+}
```

Comment:
How do we feel about dropping the `I` prefixes here?

I was thinking something like this for the overall API.

![image](https://github.com/solana-labs/solana-web3.js/assets/3642397/9331d891-31a8-40b1-93d6-6705f4cfe788)

Reply:
This API diagram looks like a good endpoint for this, I'll rename based on this.